cmd/kubectl-k8ssandra/register/register_test.go

```diff
@@ -0,0 +1,163 @@
+package register
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/k8ssandra/k8ssandra-client/internal/envtest"
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestRegister(t *testing.T) {
+	require := require.New(t)
+	client1 := (*multiEnv)[0].GetClientInNamespace("source-namespace")
+	client2 := (*multiEnv)[1].GetClientInNamespace("dest-namespace")
+	require.NoError(client1.Create((*multiEnv)[0].Context, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "source-namespace"}}))
+	require.NoError(client2.Create((*multiEnv)[1].Context, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "dest-namespace"}}))
+
+	buildDir := filepath.Join(envtest.RootDir(), "build")
+	testDir := filepath.Join(buildDir, time.Now().String())
```

Comment:
This is how it is done in k8ssandra-operator and other projects. Only cass-operator does it differently. Now, maybe that's wrong in some way here because of the way you've set up the dockerfile, but I'd actually say maybe we should be more careful about what we're building into that then.

I don't know what you mean when you say it doesn't work. The test pases, and the files are created in a dir under build/. I did find the test directory wasn't cleaned up, I've fixed that.

Reply:
Seriously, the GHA run on this PR is red smoking red. Make a wild guess which test fails and which line. The CI must pass in this PR. 

There's even a real method in Go that automatically does everything we discussed here: https://pkg.go.dev/testing#T.TempDir

Just use that, you'll remove unnecessary copy of such methods and it becomes clear to every Go coder how test handles files.  

And just because k8ssandra-operator has this weird solution which breaks the industry best practices does not mean it's correct way to do things. It's simply ignoring all ``TMPDIR`` Posix functionality, what all the other projects are doing, what git recommends (and what git does), what go recommends and best of all one of those projects you mentioned breaks its tests if you remove that build directory between tests. 

But this isn't the place to fix those projects and their odd choices (and I have no idea why such solution was ever created there).

Reply:
Mm, true. It runs locally! ☹️ And I can see why, the mkdir command in `os` obviously isn't using a `-p` option or equivalent.

I'm reluctant to use a temp directory, because these are real clusters underlying the tests, and part of the benefit is that you can dial into them easily using kubectl and the kubeconfig files they produce. I don't want this stuff sitting in a temporary directory that won't be obvious to future devs. If you want a temporary directory in the project folder (outside of ./build) then I can live with that?

Reply:
mkdir -p can be achieved with "os.MkdirAll", but there's also another issue. See the .String() return value, it has a lot of whitespace in it, since that's how time is formatted (and might be locale specific(?)).

What prevents connecting to the cluster if you use tmpDir?

Reply:
Why would the whitespace be a problem? I see no issues with having a descriptive name for the temporary folder to be honest, this would be useful to me. I've switched to using MkdirAll as you suggest. Let me know if you want a different folder to `build` used in addition. I'd rather keep everything in build, but if it matters to you then I'll change it.

Reply:
Also, having the locale listed in the timestamp is very useful for those of us who aren't in the same TZ as the github runners, so we can more easily figure out when particular test runs actually occured.

Reply:
I'm not in the same TZ as the github runners, but using UTC is for that reason the best thing ever. That way I always know when it was run, regardless of the location of the machine or locale used. But whitespace is annoying in coded environments, when we could just as well use something that's not including whitespace. 

And like the time package docs say:

"The returned string is meant for debugging; for a stable serialized representation, use t.MarshalText, t.MarshalBinary, or t.Format with an explicit format string. "

Whitespace is just super annoying in the command line.